### ISSUE TITLE
Update to 9.0 fails with General error: 1 table "oc_addressbooks" already exists

### ISSUE THREAD

**Issue by paolomonella** [owncloud/core#23667]
### Steps to reproduce
1. Update from version 8.2.2.2 to version 9.0 with OSMC on Raspberry Pi B
2. The same error occurs both when upgrading via command line (sudo -u www-data ./occ upgrade) and when upgrading via the web interface

### Expected behaviour
I expected the update to happen.

### Actual behaviour
These are the messages that I get when upgrading via command line:

These are the messages when upgrading via the web interface:

```
# sudo -u www-data ./occ upgrade
ownCloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Set log level to debug
Checking whether the database schema can be updated (this can take a long time depending on the database size)
Checked database schema update
Checking updates of apps
Checked database schema update for apps
Updating database schema
Updated database
Doctrine\DBAL\Exception\TableExistsException: An exception occurred while executing 'CREATE TABLE "oc_addressbooks" ("id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT, "principaluri" VARCHAR(255) DEFAULT NULL, "displayname" VARCHAR(255) DEFAULT NULL, "uri" VARCHAR(255) DEFAULT NULL, "description" VARCHAR(255) DEFAULT NULL, "synctoken" INTEGER UNSIGNED DEFAULT 1 NOT NULL)':

SQLSTATE[HY000]: General error: 1 table "oc_addressbooks" already exists
Update failed
Maintenance mode is kept active
Reset log level
```

These are the messages when upgrading via the web interface:

```
Updating ownCloud to version 9.0.0, this may take a while.

Preparing update
Set log level to debug
Turned on maintenance mode
Checking whether the database schema can be updated (this can take a long time depending on the database size)
Checked database schema update
Checking updates of apps
Checked database schema update for apps
Updating database schema
Updated database
Doctrine\DBAL\Exception\TableExistsException: An exception occurred while executing 'CREATE TABLE "oc_addressbooks" ("id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT, "principaluri" VARCHAR(255) DEFAULT NULL, "displayname" VARCHAR(255) DEFAULT NULL, "uri" VARCHAR(255) DEFAULT NULL, "description" VARCHAR(255) DEFAULT NULL, "synctoken" INTEGER UNSIGNED DEFAULT 1 NOT NULL)': SQLSTATE[HY000]: General error: 1 table "oc_addressbooks" already exists
The update was unsuccessful. Please report this issue to the ownCloud community.
```

### Server configuration
**Operating system**: OSMC

**Web server:** Apache/2.4.10 (Debian)

**Database:** SQLite

**PHP version:** PHP 5.6.17-0+deb8u1 (cli) (built: Jan 15 2016 17:58:43)

**ownCloud version:** (see ownCloud admin page)

**Updated from an older ownCloud or fresh install:** Updated from an older ownCloud

**Where did you install ownCloud from:** I went to https://download.owncloud.org/download/repositories/stable/owncloud/ --> I clicked on 'Debian' and followed the instructions for Debian_8.0 owncloud-8.2.2-1.1

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - dav: true
  - federatedfilesharing: 0.1.0
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - templateeditor: 0.1
Disabled:
  - calendar
  - comments
  - contacts
  - encryption
  - external
  - federation
  - files_videoplayer
  - systemtags
  - updatenotification
  - user_external
  - user_ldap
```

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'instanceid' => 'ocez6gm2m9cl',
  'passwordsalt' => '[cut]',
  'secret' => '[cut]',
  'trusted_domains' =>
  array (
    0 => '192.168.1.76',
    1 => '[cut].dyndns.org',
    2 => '100.[cut].[cut].[cut]',
  ),
  'datadirectory' => '/media/intenso/owncloud_data/',
  'overwrite.cli.url' => 'http://192.168.1.76/owncloud',
  'dbtype' => 'sqlite3',
  'version' => '8.2.2.2',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'ilbuonme',
  'dbpassword' => '[cut]',
  'logtimezone' => 'UTC',
  'installed' => true,
  'updatechecker' => false,
  'maintenance' => true,
  'theme' => '',
  'loglevel' => 2,
);
```

**Are you using external storage, if yes which one:** /media/intenso (external 1 Tb HDD, alimented through a USB hub).

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** None (as far as I know).

### Client configuration
**Browser:** Mozilla Firefox for Ubuntu 45.0

**Operating system:** Ubuntu 15.10 (Xubuntu)

#### Web server error log
```
192.168.1.65 - - [30/Mar/2016:19:00:44 +0200] "GET /core/js/files/fileinfo.js?v=3abb0f482e22568d7b5bad40877a787f HTTP/1.1" 200 1420 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
192.168.1.65 - - [30/Mar/2016:19:00:44 +0200] "GET /core/js/files/client.js?v=3abb0f482e22568d7b5bad40877a787f HTTP/1.1" 200 5222 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
192.168.1.65 - - [30/Mar/2016:19:00:44 +0200] "GET /core/vendor/jquery-ui/ui/jquery-ui.custom.js?v=3abb0f482e22568d7b5bad40877a787f HTTP/1.1" 200 106662 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
192.168.1.65 - - [30/Mar/2016:19:00:44 +0200] "GET /index.php/core/js/oc.js?v=3abb0f482e22568d7b5bad40877a787f HTTP/1.1" 200 3020 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
192.168.1.65 - - [30/Mar/2016:19:00:45 +0200] "GET /core/vendor/backbone/backbone.js?v=3abb0f482e22568d7b5bad40877a787f HTTP/1.1" 200 20651 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
192.168.1.65 - - [30/Mar/2016:19:00:45 +0200] "GET /core/img/logo-icon.svg HTTP/1.1" 304 180 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
192.168.1.65 - - [30/Mar/2016:19:00:45 +0200] "GET /core/fonts/OpenSans-Light.woff HTTP/1.1" 200 63327 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
192.168.1.65 - - [30/Mar/2016:19:00:45 +0200] "GET /core/img/breadcrumb.svg HTTP/1.1" 304 180 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
192.168.1.65 - - [30/Mar/2016:19:00:46 +0200] "GET /core/ajax/update.php?requesttoken=NEEwdGQ%2FNTpzBGJtGwkYS11HBEIHYi0BbwUGADUlSnI%3D%3AMxCL0zgQ4lIYBG%2F%2FiqQ%2Fe2a3Vm6YAg%2B1YiQwhiumGR0%3D HTTP/1.1" 200 2011 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
127.0.0.1 - - [30/Mar/2016:19:02:09 +0200] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.10 (Debian) (internal dummy connection)"
```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"Zs7fb3Kxl7yNoCOwqjqQ","remoteAddr":"192.168.1.65","app":"PHP","message":"fopen(\/var\/www\/owncloud\/config\/config.php): failed to open stream: No such file or directory at \/var\/www\/owncloud\/lib\/private\/config.php#188","level":3,"time":"2016-02-07T15:45:11+00:00"}
{"reqId":"p4N7P7orIaH4dpEGHjoD","remoteAddr":"192.168.1.65","app":"mysql.setup","message":"Specific user creation failed: An exception occurred while executing 'SELECT user FROM mysql.user WHERE user=?' with params [\"oc_ilbuonme\"]:\n\nSQLSTATE[42000]: Syntax error or access violation: 1142 SELECT command denied to user 'ilbuonme'@'localhost' for table 'user'","level":3,"time":"2016-02-07T15:50:01+00:00"}
{"reqId":"p4N7P7orIaH4dpEGHjoD","remoteAddr":"192.168.1.65","app":"mysql.setup","message":"Database creation failed: An exception occurred while executing 'GRANT ALL PRIVILEGES ON `owncloud` . * TO 'ilbuonme'':\n\nSQLSTATE[42000]: Syntax error or access violation: 1044 Access denied for user 'ilbuonme'@'localhost' to database 'owncloud'","level":3,"time":"2016-02-07T15:50:01+00:00"}
{"reqId":"p4N7P7orIaH4dpEGHjoD","remoteAddr":"192.168.1.65","app":"PHP","message":"Undefined offset: 0 at \/var\/www\/owncloud\/lib\/private\/setup\/mysql.php#45","level":3,"time":"2016-02-07T15:50:01+00:00"}
{"reqId":"p4N7P7orIaH4dpEGHjoD","remoteAddr":"192.168.1.65","app":"PHP","message":"touch(): Unable to create file \/media\/intenso\/owncloud_data\/\/ilbuonme\/files\/Photos\/Squirrel.jpg because Read-only file system at \/var\/www\/owncloud\/lib\/private\/files\/storage\/local.php#169","level":3,"time":"2016-02-07T15:50:08+00:00"}
```

#### Browser log
```
[...]
```

**Comment by paolomonella** [comment]
I continued searching through this repository, and found that maybe my issue is a duplicate of https://github.com/owncloud/core/issues/22959

If this is the case, the proposed solution would be

> Between two upgrade attempts you have to reset your DB to the 8.2.2 version.
> That means delete all tables and then restore the backup.

My problem with this is that I had not created a backup of my DB before starting the update (my fault!). Is there anything I can do now to recover my DB, or should I just say goodbye to my DB, delete all tables and then update again via command line?

**Comment by paolomonella** [comment]
I solved my own issue by deleting the old database in MariaDB, creating a new database, uninstalling and reinstalling OC via the repository. In other words, I lost my database, but it's no big deal.
